react-backend/client/src/components/admin/adminUserDetailsPage.js

```diff
@@ -0,0 +1,270 @@
+      userFullName: '',
+      sex: '',
+      yearOfBirth: '',
+      email: '',
+      signupDate: '',
+      hash: ''
+    };
+  }
+
+  setUser = user => {
+    this.setState({
+      userFullName: user.fullname,
+      sex: user.sex,
+      yearOfBirth: user.yearofbirth,
+      email: user.email,
+      signupDate: user.date
+        .split('T')
+        .join(' ')
+        .slice(0, 19),
+      hash: user.hash,
+      reauthenticated: true,
+      errorMessage: ''
+    });
+  };
+
+  headers = new Headers({
+    Authorization: localStorage.getItem('token'),
+    'Content-Type': 'application/json'
+  });
+
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value,
+      errorMessage: ''
+    });
+  };
+
+  handleChangeSex = e => {
+    this.setState({ sex: e.target.value });
+  };
+
+  handleFetchError = res => {
+    if (res.status === 500) throw new Error();
+    return res;
+  };
+
+  setErrorMessage = (error, errorString) => {
+    // console.log(error) // Uncomment to display full errors in the console.
+    this.setState({ errorMessage: errorString });
+  };
+
+  handleEmptySubmit = event => {
+    event.preventDefault();
+    this.setState({
+      errorMessage: 'Please do not leave empty input fields'
+    });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    fetch('/fetchNewUserDetails', {
+      method: 'POST',
+      headers: this.headers,
+      body: JSON.stringify({
+        userId: this.state.userId,
+        userFullName: this.state.userFullName,
+        sex: this.state.sex,
+        yearOfBirth: this.state.yearOfBirth,
+        email: this.state.email,
+        password: this.state.password
+      })
+    })
+      .then(this.handleFetchError)
+      .then(res => res.json())
+      .then(res => res.details)
+      .then(this.setUser)
+      .catch(error => {
+        this.props.history.push('/internalServerError');
+      });
+  };
+
+  authenticate = event => {
+    event.preventDefault();
+
+    fetch('/user-details', {
+      method: 'POST',
+      headers: this.headers,
+      body: JSON.stringify({
+        password: this.state.password,
+        userId: this.state.userId
+      })
+    })
+      .then(this.handleFetchError)
+      .then(res => res.json())
+      .then(res => {
+        if (res.success) {
+          return res.details;
+        } else if (res.error === 'Not logged in') {
+          throw new Error('Not logged in');
+        } else {
+          this.setState({ failure: true });
+          throw new Error('password');
+        }
+      })
+      .then(details => details[0])
+      .then(this.setUser)
+      .catch(error => {
+        if (!this.state.failure) {
+          this.props.history.push('/logincb');
+        } else if (error === '500') {
+          this.props.history.push('/internalServerError');
+        }
+      });
+  };
+
+  passwordError = <span>The password is incorrect. Please try again.</span>;
+
+  renderAuthenticated = () => {
```

Comment:
I think this component should be split into multiple functional components as you did with the activities page.

Reply:
as we agreed we leave it for now :)